airframe-metrics/jvm/src/main/scala/wvlet/airframe/metrics/TimeWindow.scala

```diff
@@ -155,32 +156,49 @@ class TimeWindowBuilder(val zone: ZoneOffset, currentTime: Option[ZonedDateTime]
 
   def yesterday = today.minus(1, ChronoUnit.DAYS)
 
-  private def parseOffset(o: String, windowUnit: TimeWindowUnit): ZonedDateTime = {
-    o match {
-      case "now" => now
-      case other =>
-        Try(TimeVector(o)) match {
-          case Success(x) =>
-            // When the offset string is time duration patterns (e.g., 0M, 0d, etc.)
-            x.timeWindowFrom(now).start
-          case Failure(e) =>
-            // When the offset string is the exact date
-            val (timeString, truncate) = if (o.endsWith(")")) {
-              (o.substring(0, o.length - 1), false)
-            } else {
-              (o, true)
+  @tailrec
+  private def parseOffset(o: String, windowUnit: TimeWindowUnit, adjustments: Seq[TimeVector] = Nil): ZonedDateTime = {
+    val pattern = s"^([^/]+)(/(.+))".r("duration", "sep", "offset")
+    pattern.findFirstMatchIn(o) match {
+      case Some(m) =>
+        val d        = m.group("duration")
+        val duration = TimeVector(d)
+        parseOffset(m.group("offset"), windowUnit, adjustments :+ duration)
+      case None =>
```

Comment:
Adding a comment like `// no more nested offsets` will be helpful to understand the meaning of this recursion.